float64 first
float64 second
float64 third
